### ISSUE TITLE
Local gems (specified with a path)

### ISSUE THREAD

**Issue by ianks** [manveru/bundix#17]
In my gemfile I have some gems that are local to my projects, ie:

```ruby
gem 'simple_operation', path: './simple_operaton'
```

Bundix will generate a gemset.nix for this (without a "source" attribute). However, `nix-shell` borks complaining there is no "source". How should I deal with this?

**Comment by sheganinans** [comment]
Any progress on this?

**Comment by wereHamster** [comment]
I was able to manually edit the gemset.nix file to add src and version fields, like so:

```
  refinerycms-cover_slideshows = {
    dependencies = ["refinerycms-core"];
    src = ./vendor/extensions/cover_slideshows;
    source = {
      type = "gem";
    };
    version = "VENDOR";
  };
```

**Comment by BrianHicks** [comment]
I'm using the modification that @wereHamster suggested, but when I get inside nix-shell, running things like `rake` breaks with this error:

```
The path `/nix/store/{snip}-gemfile-and-lockfile/vendor/gems/my_gem-0.1.0` does not exist. (Bundler::PathError)
```

This is true! That path in the store does not exist—in fact, `/nix/store/{snip}-gemfile-and-lockfile` does not have `vendor` or anything. The gem builds fine, but how can I get the Gemfile to reference the built path correctly?

---

Edit: Looks like no, and this use case is related to https://github.com/manveru/bundix/issues/10

My solution for now may have to be inlining the source for `my_gem` into my project. In this case, that may work and unblock me. 😄

**Comment by zarelit** [comment]
Is this solved by #37 ? If yes, it would be nice to have a release :-D
@manveru @nyarly

**Comment by zimbatm** [comment]
v2.4.0 released!

**Comment by zimbatm** [comment]
https://github.com/NixOS/nixpkgs/pull/51146
